### ISSUE TITLE
Behaviour change proposal for the keyboard/emoji drawer toggle

### ISSUE THREAD

**Issue by nikwen** [signalapp/Signal-Android#5939]
## Issue description

I see multiple related issues with the behaviour of the keyboard/emoji drawer toggle. The most important is #3898 in my opinion but there's also #3994 and some counterintuitive behaviour when in landscape mode. Let me summarize them below.

### Can't easily close emoji drawer in quick reply

1) Open quick reply from the notification drawer
2) Click on the emoji symbol to open the emoji drawer
3) Now to close it, I would intuitively try to click the keyboard symbol that appears instead of the emoji symbol
4) Sadly, that does not close the emoji drawer

### Emoji selector doesn't work with physical keyboards

1) Open a conversation with a physical keyboard attached
2) Click on the emoji symbol to open the emoji drawer
3) Now to close it, I would intuitively try to click the keyboard symbol that appears instead of the emoji symbol
4) Sadly, that does not close the emoji drawer

### Weird behaviour in landscape mode

1) Open a conversation in landscape mode
2) Click on the emoji symbol to open the emoji drawer
3) Now to close it, I would intuitively try to click the keyboard symbol that appears instead of the emoji symbol
4) Sadly, that does not close the emoji drawer but opens the landscape keyboard (fullscreen)

-------------

In all of these cases, the solution is hitting the back button of the phone one or two times, but here's the issue with that:

1) In quick reply that closes the software keyboard first and only when you press the back button a second time (which before investigating this I have been afraid to do because I would expect that to close the quick reply window), it closes the emoji drawer. Then you have to click into the input field again to open the software keyboard in order to continue typing.
2) With a physical keyboard attached, I would expect the back button to go back to the contact/conversation list. I would again be too afraid of my already typed message being deleted to try hitting the back button.
3) In landscape mode, I would expect the same as with the physical keyboard.

Summing up, in all cases the behaviour is quite counterintuitive in my opinion.

## How to solve this in code

Having a quick glance at the code, it seems as if all of these could be solved in a similar way.

1) First of all, hitting the keyboard/emoji drawer toggle should close the emoji drawer if it is open.
2) Then open the software keyboard if
    i) no hardware keyboard is attached **and**
    ii) not in landscape mode on a device where the software keyboard would occupy the full screen

-------------

Please point out any issues you find with this proposal. 😄

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
